paimon-api/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -1744,6 +1744,13 @@ public class CoreOptions implements Serializable {
                             "If true, it disables explicit type casting. For ex: it disables converting LONG type to INT type. "
                                     + "Users can enable this option to disable explicit type casting");
 
+    public static final ConfigOption<Boolean> SCAN_BATCH_PLAN_MOVE_FROM_JM_TO_TM_ENABLED =
+            key("scan.batch.plan.move-from-jm-to-tm.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "If true, scan plan will run on a single TM, you can enable it when Paimon source coordinator encounter with memory bottleneck in Flink batch job. This feature is not bounded with the consumer mechanism and it will change the DAG of the flink job, thus breaking checkpoint compatibility. So this option is not enabled by default and users have to consider this side effect before enabling it.");
```

Comment:
The current configuration name and description still seems complicated to me. In order to use this configuration from the current description, users have to know Flink and Paimon implementation details such as plan, jm, tm, which influences the readability of this configuration. Is there some better naming and description that could resolve this issue?

Besides, you can use Description.builder() to avoid a long line of description.

Reply:
Thanks, done.